pkg/health/server/server.go

```diff
@@ -417,7 +419,24 @@ func NewServer(config Config) (*Server, error) {
 	server.Client = cl
 	server.Server = *server.newServer(config.HealthAPISpec)
 
-	server.httpPathServer = responder.NewServer(config.HTTPPathPort)
+	var address []string
+	if option.Config.EnableIPv4 {
+		nodeIpv4 := node.GetInternalIPv4()
+		if nodeIpv4 != nil {
+			address = append(address, nodeIpv4.String())
+		}
+	}
+
+	if option.Config.EnableIPv6 {
+		nodeIpv6 := node.GetInternalIPv6()
+		if nodeIpv6 != nil {
+			address = append(address, nodeIpv6.String())
+		} else {
+			address = nil
+		}
```

Comment:
I'd prefer to avoid these `else` and leave the slice as is, in case the ipv4 address has been already appended.
Maybe we should add a `log` entry in case `option.Config.EnableIPv4 == true && nodeIpv4 == nil` or `option.Config.EnableIPv6 == true && nodeIpv6 == nil`.

Reply:
I'd go for the second option, seems more natural in terms of user expectations.